crates/human-panic/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+use report::{Method, Report};
+
+use std::borrow::Cow;
+use std::io::{Result as IoResult, Write};
+use std::panic::PanicInfo;
+use std::path::{Path, PathBuf};
+use termcolor::{BufferWriter, Color, ColorChoice, ColorSpec, WriteColor};
+
+/// A convenient metadata struct that describes a crate
+pub struct Metadata {
+    /// The crate version
+    pub version: Cow<'static, str>,
+    /// The crate name
+    pub name: Cow<'static, str>,
+    /// The list of authors of the crate
+    pub authors: Cow<'static, str>,
+    /// The URL of the crate's website
+    pub repository: Cow<'static, str>,
+}
+
+/// `human-panic` initialisation macro
+///
+/// You can either call this macro with no arguments `setup_panic!()` or
+/// with a Metadata struct, if you don't want the error message to display
+/// the values used in your `Cargo.toml` file.
+///
+/// The Metadata struct can't implement `Default` because of orphan rules, which
+/// means you need to provide all fields for initialisation.
+///
+/// ```ignore
+/// #[macro_use]
+/// extern crate human_panic;
+///
+/// setup_panic!(Metadata {
+///     name: env!("CARGO_PKG_NAME").into(),
+///     version: env!("CARGO_PKG_VERSION").into(),
+///     authors: "My Company Support <support@mycompany.com".into(),
+///     repository: "support.mycompany.com".into(),
+/// });
+/// ```
+#[macro_export]
+macro_rules! setup_panic {
+    ($meta:expr) => {
+        use std::panic::{self, PanicInfo};
+        use $crate::{handle_dump, print_msg, Metadata};
+
+        panic::set_hook(Box::new(move |info: &PanicInfo| {
+            let file_path = handle_dump(&$meta, info);
+
+            print_msg(file_path, &$meta)
+                .expect("human-panic: printing error message to console failed");
+        }));
+    };
+
+    () => {
+        use std::panic::{self, PanicInfo};
+        use $crate::{handle_dump, print_msg, Metadata};
+
+        let meta = Metadata {
+            version: env!("CARGO_PKG_VERSION").into(),
+            name: env!("CARGO_PKG_NAME").into(),
+            authors: env!("CARGO_PKG_AUTHORS").replace(":", ", ").into(),
+            repository: env!("CARGO_PKG_REPOSITORY").into(),
+        };
+
+        panic::set_hook(Box::new(move |info: &PanicInfo| {
+            let file_path = handle_dump(&meta, info);
+
+            print_msg(file_path, &meta)
+                .expect("human-panic: printing error message to console failed");
+        }));
+    };
+}
+
+/// Utility function that prints a message to our human users
+pub fn print_msg<P: AsRef<Path>>(file_path: Option<P>, meta: &Metadata) -> IoResult<()> {
+    let (_version, name, _authors, repository) =
+        (&meta.version, &meta.name, &meta.authors, &meta.repository);
+
+    let stderr = BufferWriter::stderr(ColorChoice::Auto);
+    let mut buffer = stderr.buffer();
+    buffer.set_color(ColorSpec::new().set_fg(Some(Color::Red)))?;
+
+    writeln!(
+        &mut buffer,
+        "{} had a problem and crashed. To help us diagnose the \
```

Comment:
Tiniest of nits that doesn't really matter, and definitely shouldn't block, but this prints "rover" instead of "Rover" 😄